src/Castle.Windsor/MicroKernel/DefaultKernel.cs

```diff
@@ -537,6 +537,10 @@ public IKernel Register(params IRegistration[] registrations)
 			var token = OptimizeDependencyResolution();
 			foreach (var registration in registrations)
 			{
+				if (registration is FromAssemblyDescriptor)
+				{
+					throw new ArgumentException("Assembly registrations must be appended with a BasedOnDescriptor.\nUse '.Pick()' to select all types in the assembly.");
```

Comment:
Complete nitpick. Can we improve this exception to specify the public API's for component registrations as opposed to the fluent registration types?

You know something like: 

"It looks like you tried to register all types in an assembly using `container.Register(Classes.FromThisAssembly());`. It is recommended that you at least use some kind of criteria like `container.Register(Classes.FromThisAssembly().BasedOn<IFoo>());`. If you meant to register everything, please use `container.Register(Classes.FromThisAssembly().Pick());` instead."